### ISSUE TITLE
Can you provide some examples?

### ISSUE THREAD

**Issue by Zamiell** [mafredri/go-trueskill#2]
Hello,

I have a game server (for 1v1 battles) written in Go and I wanted to add a leaderboard. So, I stumbled upon this package from a Google search. Unfortunately, there's no documentation! (And it seems like there were no other existing issues for me to glance at. I'm quite a beginner at Go, so I apologize in advance.)

1) Can you add a basic implementation example in the README file? Some examples would be great.

2) For my use-case I'd like to feed it something along the lines of:
```
Tom - 1000 rating
Mary - 1000 rating
Tom wins
```
and have the package spit out back at me:
```
Tom - 1100 rating
Mary - 900 rating
```
Is this kind of thing possible?

3) In order to function properly, does the package have to know about ALL of the players ratings? Or is the calculation only confined to the TrueSkill ratings of the two specific players?

**Comment by mafredri** [comment]
Hi @Zamiell, glad you're finding use for the package :-)!

> 1. Can you add a basic implementation example in the README file? Some examples would be great.

The [API documentation (GoDoc)](https://godoc.org/github.com/mafredri/go-trueskill) should provide some hints on how to get started, there is also an example. The README in the repo has a badge that links to the GoDoc-page, but perhaps I should include it in text as well to make it more obvious.

I've also implemented a server [`go-trueskilld`](https://github.com/mafredri/go-trueskilld) if you want to look at an actual implementation.

Let me know if you have ideas to improve the godocs!

> 2. For my use-case I'd like to feed it something along the lines of:

Yes! But you will need to track mu/sigma (and optionally *trueskill*) instead of just the rating 🙂 (see [example](https://godoc.org/github.com/mafredri/go-trueskill#example-package) in docs).

> 3. In order to function properly, does the package have to know about ALL of the players ratings? Or is the calculation only confined to the TrueSkill ratings of the two specific players?

You only need the two players you are battling against each other 😄. This is one of the reasons I chose to implement the TrueSkill algorithm in Go.
